lib/mbta.rb

```diff
@@ -0,0 +1,63 @@
+mbta = {
+        red: {
+          stations: [
+            "Alewife",
+            "Davis",
+            "Porter",
+            "Harvard",
+            "Central",
+            "Kendall/MIT",
+            "Park Street",
+            "South Station"
+          ]
+        },
+        green: {
+          stations: [
+            "Haymarket",
+            "Government Center",
+            "Park Street",
+            "Boylston",
+            "Arlington",
+            "Copley"
+          ]
+        },
+        orange: {
+          stations: [
+            "North Station",
+            "Haymarket",
+            "Park Street",
+            "State Street",
+            "Downtown Crossing",
+            "Chinatown",
+            "Tufts"
+          ]
+        }
+}
+
+def calc_index_distance(array, index_a, index_b)
+  ( array.index(index_a) - array.index(index_b) ).abs
+end
+
```

Comment:
`system` is an unnecessary abstraction here. We're only working with MBTA, so breaking the `mbta` hash into something like this might have been easier:

```
stations = { 
  red: ["Alewife", "Davis", "Porter", ... ],
  green: ["Haymarket", ... ],
  ...
}
```